plugins/package-managers/cocoapods/src/main/kotlin/Lockfile.kt

```diff
@@ -95,16 +97,16 @@ internal data class Lockfile(
     )
 }
 
-internal fun String.parseLockfile(): Lockfile {
-    val root = Yaml.default.parseToYamlNode(this).yamlMap
+internal fun File.parseLockfile(): Lockfile {
+    val root = Yaml.default.parseToYamlNode(this.readText()).yamlMap
 
     val externalSources = root.get<YamlMap>("EXTERNAL SOURCES")?.entries.orEmpty().map {
         val name = it.key.content
         val node = it.value.yamlMap
 
         val externalSource = ExternalSource(
-            path = node.get<YamlScalar>(":path")?.content,
-            podspec = node.get<YamlScalar>(":podspec")?.content
+            path = node.get<YamlScalar>(":path")?.content?.let { rawPath -> parentFile.resolve(rawPath).path },
+            podspec = node.get<YamlScalar>(":podspec")?.content?.let { rawPath -> parentFile.resolve(rawPath).path }
```

Comment:
Let's use `?.let { resolveSibling(it).path }` here and below.

Reply:
Ok but without the `it`, as it shadows the one of the enclosing `.map`.

Reply:
Ok, but maybe call the lamda arguments `path` and `pospec`, after their fields, instead of `rawPath`?